src/compiler/checker.ts

```diff
@@ -1868,7 +1868,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 const nodeLinks = getNodeLinks(node);
                 cachedResolvedSignatures.push([nodeLinks, nodeLinks.resolvedSignature] as const);
                 nodeLinks.resolvedSignature = undefined;
-                if (isFunctionLike(node)) {
+                if (isFunctionExpressionOrArrowFunction(node)) {
                     const symbolLinks = getSymbolLinks(getSymbolOfDeclaration(node));

```

Comment:
Maybe I'm wrong here, but is it the case that this could also be fixed by instead checking if the declaration has a symbol?

Reply:
> Maybe I'm wrong here

You are not 😉 

> but is it the case that this could also be fixed by instead checking if the declaration has a symbol?

It could have been fixed like that but the possibility of lacking a symbol isn't reflected by the types. I also just noticed that this kinda happened because I was using a function that was matching more types than I wanted so this just seemed like a slightly better fix.

Reply:
> the possibility of lacking a symbol isn't reflected by the types

This is an ongoing struggle; I should really resurrect my PR which attempts to make it known-optional.